### PR TITLE
[1798] Make sure an Org Name has been entered via MCB providers create

### PR DESCRIPTION
**Pull request by richrace**
### Context

Providers were created with an Organisation without a name.

### Changes proposed in this pull request

Before: Was able to create/assign Providers to organisation with an empty name
After: Must provide an Organisation name to create a new Provider

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by richrace** [comment]
[mcb providers create](https://trello.com/c/dXGRMUmo/1798-mcb-providers-create)

**Comment by benilovj** [comment]
CodeClimate is complaining about complexity; I think it's possible to fix this issue without adding more nesting:

```ruby
    def find_or_create_organisation
      finished_picking_organisation = false
      until finished_picking_organisation
        organisation_name = @cli.ask_organisation_name

        if organisation_name.blank?
          puts "Organisation name cannot be blank."
          next
        end

        organisation = Organisation.find_or_initialize_by(name: organisation_name)
        if organisation.persisted?
          finished_picking_organisation = true
        elsif organisation.new_record? && @cli.confirm_new_organisation_needed?
          organisation.save!
          finished_picking_organisation = true
        end
      end

      # connect provider to org
      provider.organisations << organisation

      organisation
    end
```

File: lib/mcb/provider_cli.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def ask_region_code
     end
 
     def ask_organisation_name
-      @cli.ask("What organisation should the new provider be added to?  ")
+      @cli.ask("What Organisation should the new provider be added to? (Enter Org Name)  ")
```
Comment (timabell) [inline lib/mcb/provider_cli.rb]: Like the new prompt. I was confused by the question before

**Comment by timabell** [comment]
Like it, caveat Jake's comment.

File: lib/mcb/provider_editor.rb [hunk 2]
```diff
@@ -88,13 +88,22 @@ def try_saving_provider
...
+          organisation = Organisation.find_or_initialize_by(name: org_name_input)
+
+          if organisation.persisted?
+            finished_picking_organisation = true
+          elsif organisation.new_record? && @cli.confirm_new_organisation_needed?
+            organisation.save!
+            finished_picking_organisation = true
+          end
+        else
+          puts "Organisation name cannot be blank."
```
Comment (defong) [inline lib/mcb/provider_editor.rb]: Just wondering should this be in the Organisation model itself as 

`validate :name presence`

Comment (richrace) [reply lib/mcb/provider_editor.rb]: Will to @benilovj suggestion and add this validation. 

So we have a nicer UX and stop accepting any potential bad data.

**Pull request merged**